Ver 0.68 cmake errors prevent building

Ubuntu 14.04.1 LTS 64 bit kernel 4.4.0-139-generic 
downloaded uncrustify-0.68.tar.gz
mkdir uncrustify-68
cd uncrustify-68
unpackaged uncrustify-0.68.tar.gz to .
mkdir build
cd build
cmake ..

Results with error follow:

-- The C compiler identification is GNU 4.8.4
-- The CXX compiler identification is GNU 4.8.4
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Performing Test CXXFLAG_Wall
-- Performing Test CXXFLAG_Wall - Success
-- Performing Test CXXFLAG_Wextra
-- Performing Test CXXFLAG_Wextra - Success
-- Performing Test CXXFLAG_Wshadow
-- Performing Test CXXFLAG_Wshadow - Success
-- Performing Test CXXFLAG_Wpointer_arith
-- Performing Test CXXFLAG_Wpointer_arith - Success
-- Performing Test CXXFLAG_Wcast_qual
-- Performing Test CXXFLAG_Wcast_qual - Success
-- Performing Test CXXFLAG_Wcast_align
-- Performing Test CXXFLAG_Wcast_align - Success
-- Performing Test CXXFLAG_Wc_11_extensions
-- Performing Test CXXFLAG_Wc_11_extensions - Failed
-- Looking for C++ include inttypes.h
-- Looking for C++ include inttypes.h - found
-- Looking for C++ include memory.h
-- Looking for C++ include memory.h - found
-- Looking for C++ include stdint.h
-- Looking for C++ include stdint.h - found
-- Looking for C++ include stdlib.h
-- Looking for C++ include stdlib.h - found
-- Looking for C++ include strings.h
-- Looking for C++ include strings.h - found
-- Looking for C++ include string.h
-- Looking for C++ include string.h - found
-- Looking for C++ include sys/stat.h
-- Looking for C++ include sys/stat.h - found
-- Looking for C++ include sys/types.h
-- Looking for C++ include sys/types.h - found
-- Looking for C++ include unistd.h
-- Looking for C++ include unistd.h - found
-- Looking for C++ include utime.h
-- Looking for C++ include utime.h - found
-- Looking for stdbool.h
-- Looking for stdbool.h - found
-- Looking for memset
-- Looking for memset - found
-- Looking for strcasecmp
-- Looking for strcasecmp - found
-- Looking for strchr
-- Looking for strchr - not found
-- Looking for strdup
-- Looking for strdup - found
-- Looking for strerror
-- Looking for strerror - found
-- Looking for strtol
-- Looking for strtol - found
-- Looking for strtoul
-- Looking for strtoul - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of _Bool
-- Check size of _Bool - done
-- Found PythonInterp: /usr/bin/python (found version "2.7.6") 
-- scripts/make_version.py exited with code 64: Unknown version control system in '/home/painter/downloads/uncrustify-68'.
As a fallback, version 'Uncrustify-0.68_f' will be used.
CMake Error at tests/CMakeLists.txt:23 (option):
  Syntax error in cmake code at
    /home/painter/downloads/uncrustify-68/tests/CMakeLists.txt:24
  when parsing string
    Create a separate CTest test for each test case;\
  \ this is slower, especially with Python 3
  syntax error, unexpected cal_SYMBOL, expecting $end (51)
-- Configuring incomplete, errors occurred!
See also "/home/painter/downloads/uncrustify-68/build/CMakeFiles/CMakeOutput.log".
See also "/home/painter/downloads/uncrustify-68/build/CMakeFiles/CMakeError.log"

CMakeError.log follows:

Performing C++ SOURCE FILE Test CXXFLAG_Wc_11_extensions failed with the following output:
Change Dir: /home/painter/downloads/uncrustify-68/build/CMakeFiles/CMakeTmp
Run Build Command:/usr/bin/make "cmTryCompileExec3369216874/fast"
/usr/bin/make -f CMakeFiles/cmTryCompileExec3369216874.dir/build.make CMakeFiles/cmTryCompileExec3369216874.dir/build
make[1]: Entering directory `/home/painter/downloads/uncrustify-68/build/CMakeFiles/CMakeTmp'
/usr/bin/cmake -E cmake_progress_report /home/painter/downloads/uncrustify-68/build/CMakeFiles/CMakeTmp/CMakeFiles 1
Building CXX object CMakeFiles/cmTryCompileExec3369216874.dir/src.cxx.o
/usr/bin/c++    -Wall -Wextra -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -DCXXFLAG_Wc_11_extensions   -Wc++11-extensions -o CMakeFiles/cmTryCompileExec3369216874.dir/src.cxx.o -c /home/painter/downloads/uncrustify-68/build/CMakeFiles/CMakeTmp/src.cxx
c++: error: unrecognized command line option '-Wc++11-extensions'
make[1]: *** [CMakeFiles/cmTryCompileExec3369216874.dir/src.cxx.o] Error 1
make[1]: Leaving directory `/home/painter/downloads/uncrustify-68/build/CMakeFiles/CMakeTmp'
make: *** [cmTryCompileExec3369216874/fast] Error 2
Source file was:
int main() { return 0;}
Determining if the strchr exist failed with the following output:
Change Dir: /home/painter/downloads/uncrustify-68/build/CMakeFiles/CMakeTmp
Run Build Command:/usr/bin/make "cmTryCompileExec958173503/fast"
/usr/bin/make -f CMakeFiles/cmTryCompileExec958173503.dir/build.make CMakeFiles/cmTryCompileExec958173503.dir/build
make[1]: Entering directory `/home/painter/downloads/uncrustify-68/build/CMakeFiles/CMakeTmp'
/usr/bin/cmake -E cmake_progress_report /home/painter/downloads/uncrustify-68/build/CMakeFiles/CMakeTmp/CMakeFiles 1
Building CXX object CMakeFiles/cmTryCompileExec958173503.dir/CheckSymbolExists.cxx.o
/usr/bin/c++    -Wall -Wextra -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wno-missing-field-initializers    -o CMakeFiles/cmTryCompileExec958173503.dir/CheckSymbolExists.cxx.o -c /home/painter/downloads/uncrustify-68/build/CMakeFiles/CMakeTmp/CheckSymbolExists.cxx
/home/painter/downloads/uncrustify-68/build/CMakeFiles/CMakeTmp/CheckSymbolExists.cxx: In function ‘int main(int, char**)’:
/home/painter/downloads/uncrustify-68/build/CMakeFiles/CMakeTmp/CheckSymbolExists.cxx:17:25: error: address of overloaded function with no contextual type information
   return ((int*)(&strchr))[argc];
                         ^
make[1]: Leaving directory `/home/painter/downloads/uncrustify-68/build/CMakeFiles/CMakeTmp'
make[1]: *** [CMakeFiles/cmTryCompileExec958173503.dir/CheckSymbolExists.cxx.o] Error 1
make: *** [cmTryCompileExec958173503/fast] Error 2
File /home/painter/downloads/uncrustify-68/build/CMakeFiles/CMakeTmp/CheckSymbolExists.cxx:
/* */
#include <inttypes.h>
#include <memory.h>
#include <stdint.h>
#include <stdlib.h>
#include <strings.h>
#include <string.h>
#include <sys/stat.h>
#include <sys/types.h>
#include <unistd.h>
#include <utime.h>
int main(int argc, char** argv)
{
  (void)argv;
#ifndef strchr
  return ((int*)(&strchr))[argc];
#else
  (void)argc;
  return 0;
#endif
}

---------------------------
regards
oldunixguy

Strange..... I edited the CMakeLists.txt in tests.... the passage now reads:
option("UNCRUSTIFY_SEPARATE_TESTS"
  "Create a separate CTest test for each test case"
  " this is slower, especially with Python 3" OFF
)

YET not working.... "cd build; cmake .." outputs to the terminal:
-- scripts/make_version.py exited with code 64: Unknown version control
system in '/home/painter/downloads/uncrustify-68'.

As a fallback, version 'Uncrustify-0.68_f' will be used.
CMake Error at tests/CMakeLists.txt:23 (option):
  option called with incorrect number of arguments:
UNCRUSTIFY_SEPARATE_TESTS
  Create a separate CTest test for each test case this is slower, especially
  with Python 3 OFF
-- Configuring incomplete, errors occurred!
See also
"/home/painter/downloads/uncrustify-68/build/CMakeFiles/CMakeOutput.log".
See also
"/home/painter/downloads/uncrustify-68/build/CMakeFiles/CMakeError.log".




On Mon, Nov 19, 2018 at 8:23 AM Matthew Woehlke <notifications@github.com>
wrote:

> My guess is it's a CMake version thing... @oldunixguy
> <https://github.com/oldunixguy>, what version of CMake are you using?
>
> Does this patch fix the problem?
>
> diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
> index f54f339d..1cdf0928 100644--- a/tests/CMakeLists.txt+++ b/tests/CMakeLists.txt@@ -20,10 +20,11 @@ if (NOT ENABLE_CODECOVERAGE)
>    list(APPEND test_suites staging.test)
>  endif()
>  -option("UNCRUSTIFY_SEPARATE_TESTS"-  "Create a separate CTest test for each test case;\-\ this is slower, especially with Python 3" OFF+set(stdoc+  "Create a separate CTest test for each test case"+  " this is slower, especially with Python 3"
>  )+option("UNCRUSTIFY_SEPARATE_TESTS" "${stdoc}" OFF)
>  if (UNCRUSTIFY_SEPARATE_TESTS)
>    set(tests_ctest_file "${CMAKE_CURRENT_BINARY_DIR}/tests.cmake")
>    set_property(DIRECTORY PROPERTY TEST_INCLUDE_FILE ${tests_ctest_file})
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/uncrustify/uncrustify/issues/2082#issuecomment-439930181>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMMAj0F1qj9m41L3TamAPhKAo_YnVNojks5uws0KgaJpZM4Yno2R>
> .
>


-- 
Richard A. Painter, P.E.

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -20,10 +20,11 @@ if (NOT ENABLE_CODECOVERAGE)
   list(APPEND test_suites staging.test)
 endif()
 
-option("UNCRUSTIFY_SEPARATE_TESTS"
-  "Create a separate CTest test for each test case;\
-\ this is slower, especially with Python 3" OFF
+set(stdoc
+  "Create a separate CTest test for each test case"
+  " this is slower, especially with Python 3"
 )
+option("UNCRUSTIFY_SEPARATE_TESTS" "${stdoc}" OFF)
 if (UNCRUSTIFY_SEPARATE_TESTS)
   set(tests_ctest_file "${CMAKE_CURRENT_BINARY_DIR}/tests.cmake")
   set_property(DIRECTORY PROPERTY TEST_INCLUDE_FILE ${tests_ctest_file})

